packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -352,4 +353,21 @@ export class SaveFileDialog extends FileDialog<URI | undefined> {
         this.fileNameField.onkeyup = () => this.validate();
     }
 
+    protected async accept(): Promise<void> {
+        if (await this.confirmOverwrite()) {
+            super.accept();
+        }
+    }
+
+    private async confirmOverwrite(): Promise<boolean> {
+        const value = this.value;
+        if (value && await this.fileSystem.exists(value.toString())) {
+            const confirmed = await new ConfirmDialog({
```

Comment:
Should not be the check be done by clients only when it is required?

Reply:
all humans make mistakes, we don't want to mistakenly overwrite an existing file that has important information, do we :)

Reply:
>I think it is up to the client to decide how to save, whether always overwrite, ask, maybe to propose an alternative path or create some files in a selected folder. I don't see a reason to lock-in the save dialog just to one scenario.

you are right. i will revert the change made to the dialog itself. 

>BTW why do you do it, I could not find a single client?

it is used in one change I am working on. i prefer to keep them apart, to make PRs more specific.

Reply:
@akosyakov done.